lib/autoload/coursemology.rb

```diff
@@ -0,0 +1,7 @@
+module Coursemology
+  class ComponentNotFoundError < StandardError
```

Comment:
Then putting it in the top level module doesn't serve any purpose. Place it in the component host or something.